### PR TITLE
add .each method to calendar

### LINKED ISSUE #14: Implement Calendar#each
Iterating over days of the liturgical year, yielding `Day` instances.

### PR DESCRIPTION
**Pull request by lao9**
closes #14 

@igneus - please let me know if you'd like any modifications to anything! it seems simple, but took me some time to orient myself to the repo.

### CODE REVIEW

File: spec/calendar_spec.rb [hunk 1]
```diff
@@ -33,6 +33,14 @@
     end
   end
 
+  describe '.each' do
+    it 'yields calendar day instances' do
```
Comment (igneus) [inline spec/calendar_spec.rb]: Please add another test checking that `each` actually yields. This one only checks what it yields *if* it yields.

File: lib/calendarium-romanum/calendar.rb [hunk 2]
```diff
@@ -126,6 +126,12 @@ def day(*args, vespers: false)
       )
     end
 
+    def each
+      (temporale.start_date..temporale.end_date)
+        .map { |date| day(date) }
```
Comment (igneus) [inline lib/calendarium-romanum/calendar.rb]: Both time and memory are being wasted here:

* `map` iterates over the date range, retrieves calendar data and collects them in a new array
* `each` iterates over this array and yields it's elements

Both (retrieve calendar data, yield them) should be done in a single `each`. This way you save one iteration as well as memory necessary for the temporary array.

**Comment by lao9** [comment]
Hi @igneus - I really appreciate your feedback. Thanks for your patience. This is my first time using some of rspec's yield-matchers! Please review and let me know if you have any further feedback.

**Comment by igneus** [comment]
Thanks, now everything is OK. Merging.

**Pull request merged**